CMakeLists.txt

```diff
@@ -107,15 +107,15 @@ ENDIF(NOT DEFINED CMAKE_Fortran_COMPILER_SUPPORTS_F08)
 #OPTION(USE_MPI "Use the MPI library for parallelization" OFF)
 #OPTION(USE_OPENMP "Use OpenMP for parallelization" OFF)
 
-IF(CMAKE_Fortran_COMPILER_ID MATCHES "Cray")
+IF(CMAKE_Fortran_COMPILER_ID MATCHES "(Cray|LLVMFlang)")
   set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} ${HIPFORT_COMPILER_FLAGS}")
-ELSE(CMAKE_Fortran_COMPILER_ID MATCHES "Cray")
+ELSE(CMAKE_Fortran_COMPILER_ID MATCHES "(Cray|LLVMFlang)")
 IF(CMAKE_Fortran_COMPILER_SUPPORTS_F08)
   set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} ${HIPFORT_COMPILER_FLAGS} -std=f2008")
 ELSE(CMAKE_Fortran_COMPILER_SUPPORTS_F08)
   set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} ${HIPFORT_COMPILER_FLAGS} -std=f2003")
 ENDIF(CMAKE_Fortran_COMPILER_SUPPORTS_F08)
-ENDIF(CMAKE_Fortran_COMPILER_ID MATCHES "Cray")
+ENDIF(CMAKE_Fortran_COMPILER_ID MATCHES "(Cray|LLVMFlang)")
```

Comment:
It's a bit surprising that there's no `Fortran_STANDARD` target property. I wonder why that is...

This should probably be replaced with `try_compile` or something of that sort, but I'm certainly not going to hold you to a higher standard than the original author.

Reply:
https://gitlab.kitware.com/cmake/cmake/-/issues/25006
Seems like the target property would make sense. In general though the `-std` flags only restrict the use of newer standard features rather than enable features from newer standards, so they really aren't necessary unless you want to ensure certain files to conform to older standards.